stubs/tifftools/tifftools.pyi

```diff
@@ -27,7 +27,7 @@ class IFD(TypedDict):
 class TagEntry(TypedDict):
     datatype: int
     count: int
-    datapos: int
+    datapos: NotRequired[int]
```

Comment:
It seems like this is actually always returned: https://github.com/DigitalSlideArchive/tifftools/blob/7b94211693800f997205b1d1225ef63ae894e29c/tifftools/tifftools.py#L183

When is it missing?

Reply:
`datapos` is not required for writing.   You could copy them when copying the tag (it will do no harm).  When combining ifds from two different files, tifftools has some bookkeeping about what file the data came from and the `StripOffsets` / `TileOffsets` (or other data blocks that have offsets and bytecount tags) use that to read the appropriate parts from the different source files.  If you are copying those tags to a new/different ifd, then you must also copy the `path_or_fobj` entry that is in the ifd so it knows where the data is being read from.

Reply:
From a type annotation perspective, we should not make something `NotRequired` if an API guarantees that it'll be returned, as it forces us to implement unnecessary additional defensive code when using the API.

Maybe you need to split `TagEntry` into `ReadTagEntry` and `WriteTagEntry` classes, then annotate each part (places for reading vs writing) of the `tifftools` API with the appropriate object.

Reply:
I'll revert this change since I'm trying a different approach to preserve metadata when creating a new image. If that approach doesn't end up working then I can look into splitting `TagEntry`.